pkg/controller/installation/core_controller.go

```diff
@@ -1472,102 +1385,53 @@ func (r *ReconcileInstallation) SetDegraded(reason string, err error, log logr.L
 	r.status.SetDegraded(reason, err.Error())
 }
 
-// GetTyphaNodeTLSConfig reads and validates the CA ConfigMap and Secrets for
+// GetOrCreateTyphaNodeTLSConfig reads and validates the CA ConfigMap and Secrets for
 // Typha and Felix configuration. It returns the validated resources or error
 // if there was one.
-func (r *ReconcileInstallation) GetTyphaNodeTLSConfig() (*render.TyphaNodeTLS, error) {
+func GetOrCreateTyphaNodeTLSConfig(cli client.Client, certificateManager certificatemanagement.CertificateManager) (*render.TyphaNodeTLS, error) {
 	// accumulate all the error messages so all problems with the certs
 	// and CA are reported.
-	errMsgs := []string{}
-	ca, err := r.validateTyphaCAConfigMap()
-	if err != nil {
-		errMsgs = append(errMsgs, fmt.Sprintf("CA for Typha is invalid: %s", err))
-	}
-
-	node, err := utils.ValidateCertPair(
-		r.client,
-		common.OperatorNamespace(),
-		render.NodeTLSSecretName,
-		render.TLSSecretKeyName,
-		render.TLSSecretCertName,
-	)
-	if err != nil {
-		errMsgs = append(errMsgs, fmt.Sprintf("CertPair for Felix is invalid: %s", err))
-	} else if node != nil {
-		if node.Data != nil {
-			// We need the CommonName, URISAN, or both to be set
-			_, okCN := node.Data[render.CommonName]
-			_, okUS := node.Data[render.URISAN]
-			if !(okCN || okUS) {
-				errMsgs = append(errMsgs, fmt.Sprintf("CertPair for Felix does not contain common-name or uri-san"))
+	var errMsgs []string
+	getKeyPair := func(secretName, commonName string) (keyPair certificatemanagement.KeyPair, cn string, uriSAN string) {
```

Comment:
What's the benefit of having this as anonymous as opposed to a private function?